### PR TITLE
fix: remove &read_guestbook=true from url on settings update

### PR DESCRIPTION
**Pull request by daianaszwimer**
since there's no way to send params to `redirect_back`, the change is on the client side

### CODE REVIEW

**Comment by antsgar** [comment]
If we're only using these endpoints from Settings, we could change redirect_back to redirect_to_authenticated_settings to fix this on the server side

**Comment by daianaszwimer** [comment]
fixed! @antsgar

**Comment by atmoio** [comment]
(Will defer to @antsgar on this one)

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def set_meta_images_for_letter(first_letter)
   end
 
   def redirect_to_authenticated_settings(author)
-    redirect_to "#{ENV['HOST']}/authors/#{author.id}/settings/?secret=#{author.secret}", :status => 303
+    redirect_to "#{author.url}/authors/#{author.id}/settings/?secret=#{author.secret}", :status => 303
```
Comment (antsgar) [inline app/controllers/application_controller.rb]: Shouldn't this url be "#{author.url}/settings/?secret=#{author.secret}"?

Comment (daianaszwimer) [reply app/controllers/application_controller.rb]: Nope, the settings url is /authors/id/settings :)

Comment (antsgar) [reply app/controllers/application_controller.rb]: author.url already maps to authors/[id] (or /@username or custom domain). This url for an author without a username will map to authors/[id]/authors/[id]

Comment (antsgar) [reply app/controllers/application_controller.rb]: Perhaps use author.get_host instead of author.url?

Comment (daianaszwimer) [reply app/controllers/application_controller.rb]: nice catch! @antsgar , fixed!

**Review by antsgar — APPROVED** [review]

**Pull request merged**